include/deal.II/hp/refinement.h

```diff
@@ -19,10 +19,26 @@
 
 #include <deal.II/base/config.h>
 
-#include <vector>
+#include <functional>
+
+#ifdef DEAL_II_WITH_CXX20
```

Comment:
This is a feature since `C++20`. There is a `boost` equivalent.

Do we have already have a `DEAL_II_WITH_CXX20` flag?

Reply:
We could also avoid the additional namespace by putting another `#ifdef` block around the `alias`.

Reply:
@masterleinad -- FYI

Reply:
We already have `identity` in ` include/deal.II/base/template_constraints.h` that does exactly the same.
I currently don't see the need for using the C++20 version instead.

No, we don't detect C++20 yet.

Reply:
Use the class we provide, rather than the C++20 class. We use it in many contexts already -- just search for `identity<` :-)

Reply:
Thanks to both of you! I suspected you were referring to the `std` functionality and didn't know about the `deal.II` one yet.

Is there any difference between `const identity<T>::type &` and `identity<const T&>`? I would use the former variant since it is more commonly used in `deal.II`.